Packs/Cofense-Intelligence/Integrations/CofenseIntelligenceV2/CofenseIntelligenceV2.py

```diff
@@ -0,0 +1,390 @@
+
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    Raises exceptions if something goes wrong.
+
+    :type client: ``Client``
+    :param Client: client to use
+
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``str``
+    """
+
+    message: str = ''
+    try:
+        client.threat_search_call()
+        message = 'ok'
+    except DemistoException as e:
+        if 'Forbidden' in str(e) or 'Authorization' in str(e):
+            message = 'Authorization Error: make sure user name and password are correctly set'
+        else:
+            raise e
+    return message
+
+
+def search_url_command(client: Client, args: Dict[str, Any], params) -> CommandResults:
+    url = args.get('url')
+    if not url:
+        raise ValueError('url not specified')
+    result = client.threat_search_call(url=url)
+
+    threats = result.get('data', {}).get('threats', [])
+    md_data, dbot_score = threats_analysis(threats, indicator=url, threshold=params.get('url_threshold'))
+    result = client.threat_search_call(url=url)
+
+    dbot_score_obj = Common.DBotScore(indicator=url, indicator_type=DBotScoreType.URL,
+                                      integration_name=INTEGRATION_NAME, score=dbot_score,
+                                      reliability=params.get(RELIABILITY))
+    url_indicator = Common.URL(url=url, dbot_score=dbot_score_obj)
+
+    return CommandResults(
+        outputs_prefix=OUTPUT_PREFIX,
+        outputs_key_field='id',
+        outputs=threats,
+        raw_response=result,
+        readable_output=tableToMarkdown(name=f'Cofense URL Reputation for url {url}', t=md_data,
+                                        headers=['Threat ID', 'Threat Types', 'Verdict', 'Executive Summary',
+                                                 'Campaign', 'Last Published']),
+        indicator=url_indicator,
+
+    )
+
+
+def check_ip_command(client: Client, args: Dict[str, Any], params) -> CommandResults:
+    ip = args.get('ip')
+    if not ip:
+        raise ValueError('IP not specified')
+    try:
+        socket.inet_aton(ip)
+        # legal
+    except socket.error:
+        raise ValueError('Invalid IP')
+
+    # Call the Client function and get the raw response
+    result = client.threat_search_call(ip=ip)
+    threats = result.get('data', {}).get('threats', [])
+    indicator_found = False
+    threshold = params.get('ip_threshold')
+    threshold_score = SEVERITY_SCORE.get(threshold)
+    if not threshold_score:
+        raise Exception(
+            f'Cofense error: Invalid threshold value: {threshold}. Valid values are: None, Minor, Moderate or Major')
+
+    dbot_score = 0
+    dbot_score_obj = Common.DBotScore(indicator=ip, indicator_type=DBotScoreType.IP,
+                                      integration_name=INTEGRATION_NAME, score=dbot_score,
+                                      reliability=params.get(RELIABILITY))
```

Comment:
why are you doing this here and at the bottom? seems redundant as you might change the score with the processing you are doing later on

Reply:
Right, fixed that